ui/app/routes/vault/cluster/secrets/backend/configuration/index.js

```diff
@@ -124,6 +126,35 @@ export default class SecretsBackendConfigurationRoute extends Route {
     }
   }
 
+  async fetchAzureConfig(id) {
+    try {
+      let azureModel = await this.store.queryRecord('azure/config', { backend: id });
+      let issuer = null;
+      if (azureModel.isConfigured) {
+        if (this.version.isEnterprise) {
+          // Issuer is an enterprise only related feature
+          // Issuer is also a global endpoint that doesn't mean anything in the Azure secret details context if WIF related fields on the azureConfig have not been set.
+          const WIF_FIELDS = ['identityTokenAudience', 'identityTokenTtl'];
+          WIF_FIELDS.some((field) => azureModel[field]) ? (issuer = await this.fetchIssuer()) : null;
+        }
+      } else {
+        // azure will return a 200 if the config is set or not set.
+        // thus, we set the model to null if no params have been configured.
+        azureModel = null;
+      }
+
+      const configArray = [];
+      configArray.push(azureModel, issuer);
+      return configArray;
```

Comment:
Is there a reason for including null values in the array? It feels a little strange to push a `null` value and then iterate over that in the template 🤔 I'd expect the types to stay consistent here and have `configArray` either be an empty array or an array of models. 

```suggestion
      const configArray = [];
      if (azureModel) configArray.push(azureModel);
      if (issuer) configArray.push(issuer);
      return configArray;
```

Reply:
Good idea. There is one change had to make to this, which is to check isConfigured on the model. The model always returns unless a true error occurs. And here I'm more or less trying to tell the template if it's configured or not by sending in null. However, I agree, pushing null is weird. See updated code.